importer/UyEs.py

```diff
@@ -0,0 +1,114 @@
+from Monument import Monument, Dataset
+import importer_utils as utils
+import importer as importer
+
+
+class UyEs(Monument):
+
+    def set_inception(self):
+        """Set building year if parseable."""
+        if self.has_non_empty_attribute("construido"):
+            if utils.legit_year(self.construido):
+                inc_year = {"time_value": {"year": self.construido}}
+                self.add_statement("inception", inc_year)
+            else:
+                self.add_to_report("construido", self.construido, "inception")
+
+    def set_location(self):
+        """
+        Set the Location.
+
+        Use the linked Localidad if available,
+        and if it's not linked, try and see if there's
+        an article anyway. Compare against external
+        list of settlements.
+        """
+        loc_dic = self.data_files["settlements"]
+        loc_q = None
+
+        if self.has_non_empty_attribute("localidad"):
+            loc_raw = self.localidad
+            if utils.count_wikilinks(loc_raw) == 1:
+                loc_try = utils.q_from_first_wikilink("es", loc_raw)
+                if loc_try in [x["item"] for x in loc_dic]:
+                    loc_q = loc_try
+            else:
+                loc_try = utils.q_from_wikipedia("es", loc_raw)
+                if loc_try in [x["item"] for x in loc_dic]:
+                    loc_q = loc_try
+
+            if loc_q:
+                self.add_statement("location", loc_q)
+            else:
+                self.add_to_report("localidad", self.localidad, "location")
+
+    def set_adm_location(self):
+        """Set the Admin Location using department ISO code."""
+        dep_dic = self.data_files["departments"]
+        iso = self.dep_iso
+        dep_match = utils.get_item_from_dict_by_key(
+            dict_name=dep_dic,
+            search_in="iso",
+            search_term=iso)
+        if len(dep_match) == 1:
+            adm_q = dep_match[0]
+
+        if adm_q:
+            self.add_statement("located_adm", adm_q)
+        else:
+            self.add_to_report("dep_iso", self.dep_iso, "located_adm")
+
+    def set_wd_item_via_name(self):
+        """
+        Attemt to set wd item via name.
+
+        There's no monument_article field in this dataset,
+        so try to get the Q of an article with the
+        same name, if available.
+        """
+        if utils.count_wikilinks(self.monumento) == 1:
+            monum_q = utils.q_from_first_wikilink("es", self.monumento)
```

Comment:
The same mechanism was implemented for Georgia as part of #128

Reply:
Since I'll revert that change for Georgia (due to low data quality) I've put it below instead:
```python
    def set_wd_item_via_name(self):
        """
        Attempt to set wd item via name.

        Populates self.monument_article, the value then gets used in
        find_matching_wikidata() with extra precautionary logic.
        """
        if utils.count_wikilinks(self.name) == 1:
            self.monument_article = utils.get_wikilinks(self.name)[0].title
```

Reply:
Done